service/mongodb/controller.go

```diff
@@ -163,19 +164,54 @@ func (c Controller) GetFilePermissions(ctx context.Context,
 // otherwise returns nil and any error if occurred.
 func (c Controller) GetUserPermissions(
 	ctx context.Context,
-	userID string) ([]*pb.GetUserPermissionsResponse_FileRole, error) {
-	filter := bson.D{
+	userID string, pageNum int64, pageSize int64, isShared bool, appID string) (*pb.GetUserPermissionsResponse, error) {
+
+	var filter bson.D
+
+	filter = append(filter, bson.E{
+		Key:   PermissionBSONUserIDField,
+		Value: userID,
+	})
+
+	if isShared {
+		filter = append(filter, bson.E{
+			Key:   PermissionBSONCreatorField,
+			Value: bson.M{"$ne": userID},
+		})
+	}
+
+	if appID != "" {
+		filter = append(filter, bson.E{
+			Key:   PermissionBSONAppIDField,
+			Value: appID,
+		})
+	}
+
+	// Check if one is negative and the other is not
+	if pageNum < 0 || pageSize < 0 {
```

Comment:
I think that you should check function parameters at the begging of the function.

Reply:
Correct. done.